app/views/admin/loans/_loans_grid.html.slim

```diff
@@ -36,13 +36,20 @@
       - number_to_currency(loan.amount)
     - g.column name: t('activerecord.attributes.loan.currency'), attribute: 'symbol',
       assoc: :currency, filter: false, in_html: false
-    
-    - g.column name: t('activerecord.attributes.loan.primary_agent'), attribute: 'name',
-      assoc: :primary_agent, filter: false, in_html: false
-    - g.column name: t('activerecord.attributes.loan.secondary_agent'), attribute: 'name',
-      assoc: :secondary_agent, filter: false, in_html: false
-    - g.column name: t('activerecord.attributes.loan.representative'), attribute: 'name',
-      assoc: :representative, filter: false, in_html: false
+
+    - g.column name: t('activerecord.attributes.loan.primary_agent'), in_csv: false, attribute: 'name', assoc: :primary_agent,
+      filter: true, in_html: true, filter_type: :custom_agent_filter, detach_with_id: :agent_filter do |loan|
+      - primary_name = loan.primary_agent ? loan.primary_agent.name : ""
+      - secondary_name = loan.secondary_agent ? ", #{loan.secondary_agent.name}" : ""
+      - primary_name + secondary_name
```

Comment:
This is a bit awkward and too much logic for a view I think. Please move this into the model as a method called `agent_names` that returns an array with nils removed. Then you can just do `agent_names.join(", ")` here.